Add PEP 613 TypeAlias to typing_extensions

east825 opened the pull request.

gvanrossum commented on typing_extensions/src_py2/test_typing_extensions.py:
```diff
@@ -377,6 +377,44 @@ def test_annotated_in_other_types(self):
         self.assertEqual(X[int], List[Annotated[int, 5]])
 
 
+class TypeAliasTests(BaseTestCase):
```
Could you add some positive tests? I've forgotten the syntax we agreed upon and there's no hint in this PR of what it should be.

gvanrossum commented on typing_extensions/src_py2/typing_extensions.py:
```diff
@@ -241,5 +241,43 @@ class Annotated(object):
+
+        Predicate = Callable[[...], bool]  # type: TypeAlias
+
+    It's invalid when used anyhow but as in the example above.
```
anyhow but -> anywhere except

gvanrossum commented on typing_extensions/src_py3/typing_extensions.py:
```diff
@@ -2071,3 +2071,100 @@ def get_args(tp):
+
+            Predicate: TypeAlias = Callable[[...], bool]
+
+        It's invalid when used anyhow but as in the example above.
```
anyhow as -> anywhere except

gvanrossum commented on typing_extensions/src_py3/typing_extensions.py:
```diff
@@ -2071,3 +2071,100 @@ def get_args(tp):
+
+            Predicate: TypeAlias = Callable[[...], bool]
+
+        It's invalid when used anyhow but as in the example above.
```
(again)

gvanrossum commented on typing_extensions/src_py3/typing_extensions.py:
```diff
@@ -2071,3 +2071,100 @@ def get_args(tp):
+
+            Predicate: TypeAlias = Callable[[...], bool]
+
+        It's invalid when used anyhow but as in the example above.
```
(ditto)

east825 commented:
Oh, sorry, I missed the notifications. 

I've added positive tests and fixed docstrings as suggested. Also, there happened to be a mistake in `Callable` syntax (shame on me) and a few code style inconsistencies that I overlooked initially. 

I pushed everything as individual commits to make the changes more evident as e.g. typeshed contributing guidelines require, let me know if I need to squash them manually.

gvanrossum commented:
@east825 Please don't squash, we do that when we merge. I am behind on code reviews, ping me later this week.
